index.js

```diff
@@ -11,7 +11,7 @@ module.exports = function dashify(str) {
   if (typeof str !== 'string') {
     throw new TypeError('dashify expects a string.');
   }
-  str = str.replace(/[A-Z]/g, '-$&');
+  str = str.replace(/([a-z]{1})([A-Z])/g, '$1-$2');
```

Comment:
great thanks. we don't need the `{1}` though, just the `[a-z]` should be enough.